src/api_server/src/lib.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2022 by Rivos Inc.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+//! KBS API server
+
+extern crate actix_web;
+extern crate anyhow;
+extern crate attestation_service;
+extern crate base64;
+extern crate env_logger;
+extern crate kbs_types;
+extern crate log;
+extern crate rand;
+extern crate uuid;
+
+use actix_web::{middleware, web, App, HttpServer};
+use anyhow::Result;
+
+use crate::session::SessionMap;
+
+mod http;
+mod session;
+
+/// Start the HTTP server and serve API requests.
+pub async fn serve() -> Result<()> {
+    log::info!("starting HTTP server at http://localhost:8080");
+
+    HttpServer::new(|| {
```

Comment:
A small question: How do we set HTTPS instead of HTTP?

Reply:
You pass a TLS builder to the actix app constructor, something like that:
https://actix.rs/docs/server/#tls--https

So we will also need to get the mandatory cert chain and private key from the command line as well. Eventually we could have a configuration file for the KBS and take those pieces of information (IP, port, cert, key) from there.